domain/model/state/modelstate.go

```diff
@@ -1060,3 +1060,11 @@ func (s *ModelState) IsControllerModel(ctx context.Context) (bool, error) {
 
 	return m.IsControllerModel, nil
 }
+
+// HasValidCredential returns true if the model has a valid credential.
+// The following errors may be returned:
+// - [modelerrors.NotFound] when the model no longer exists.
+func (s *ModelState) HasValidCredential(ctx context.Context) (bool, error) {
```

Comment:
you will only be able to answer this question from the controllers state. We don't have this available in the model. This will also mean putting back in the controller state to the service sorry.

Reply:
Hi tom, I think this has already been refactored to the controller state in the model domain, perhaps this review is for an older commit?